gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALRasterSource.scala

```diff
@@ -33,31 +33,31 @@ case class GDALRasterSource(
 ) extends RasterSource {
   val path: String = dataPath.path
 
-  lazy val datasetType: Int = options.datasetType
+  @transient lazy val datasetType: Int = options.datasetType
 
   // current dataset
   @transient lazy val dataset: GDALDataset =
     GDALDataset(path, options.toWarpOptionsList.toArray)
 
-  lazy val bandCount: Int = dataset.bandCount
+  @transient lazy val bandCount: Int = dataset.bandCount
 
-  lazy val crs: CRS = dataset.crs
+  @transient lazy val crs: CRS = dataset.crs
 
   // noDataValue from the previous step
-  lazy val noDataValue: Option[Double] = dataset.noDataValue(GDALWarp.SOURCE)
+  @transient lazy val noDataValue: Option[Double] = dataset.noDataValue(GDALWarp.SOURCE)
 
-  lazy val dataType: Int = dataset.dataType
+  @transient lazy val dataType: Int = dataset.dataType
 
-  lazy val cellType: CellType = dstCellType.getOrElse(dataset.cellType)
+  @transient lazy val cellType: CellType = dstCellType.getOrElse(dataset.cellType)
 
-  lazy val gridExtent: GridExtent[Long] = dataset.rasterExtent(datasetType).toGridType[Long]
+  @transient lazy val gridExtent: GridExtent[Long] = dataset.rasterExtent(datasetType).toGridType[Long]
 
   /** Resolutions of available overviews in GDAL Dataset
     *
     * These resolutions could represent actual overview as seen in source file
     * or overviews of VRT that was created as result of resample operations.
     */
-  lazy val resolutions: List[GridExtent[Long]] = dataset.resolutions(datasetType).map(_.toGridType[Long])
+  @transient lazy val resolutions: List[GridExtent[Long]] = dataset.resolutions(datasetType).map(_.toGridType[Long])
```

Comment:
For instance, why this is required to be `transient`? it is a serializable value. I would like to see why these values becoming `nulls` after deserialization.

Reply:
Sine `@transient`sets value to `null` during serialization, it is not obvious how it addresses the behavior that is described in #82, since it will cause references to be set to `nulls`. Check out for instance [this source](https://dzone.com/articles/diving-deeper-into-the-java-transient-modifier) to learn more about how `transient` notation works.